File: storm-java21/src/test/java/st/orm/template/model/OwnerView.java
package st.orm.template.model;

import org.jspecify.annotations.Nullable;
import st.orm.DbTable;
import st.orm.PK;
import st.orm.Projection;
import st.orm.Version;

@DbTable("owner_view")
public record OwnerView(
        @PK Integer id,
        String firstName,
        String lastName,
        Address address,
        @Nullable String telephone,
        @Version int version
) implements Projection<Integer> {}

File: storm-kotlin/src/test/kotlin/st/orm/template/model/OwnerView.kt
package st.orm.template.model

import st.orm.DbTable
import st.orm.PK
import st.orm.Projection
import st.orm.Version

@DbTable("owner_view")
data class OwnerView(
    @PK val id: Int = 0,
    val firstName: String,
    val lastName: String,
    val address: Address,
    val telephone: String?,
    @Version val version: Int
) : Projection<Int>
